### PR TITLE
Connect sign in form with backend

### PR DESCRIPTION
**Pull request by brovarenko**
See - https://app.asana.com/0/1202525679663009/1202782033632767/f

### CODE REVIEW

File: frontend/src/common/types/form/form-control-values.type.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { FieldValues } from 'react-hook-form';
+
+type FormControlValues = FieldValues;
+
+export { type FormControlValues };
```
Comment (oleksandr-danylchenko) [inline frontend/src/common/types/form/form-control-values.type.ts]: Why did you reexport this type and didn't use the `'react-hook-form`'s one?

File: frontend/src/common/types/form/form-control-path.type.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { Path } from 'react-hook-form';
+
+import { FormControlValues } from './form-control-values.type';
+
+type FormControlPath<T = FormControlValues> = Path<T>;
+
+export { type FormControlPath };
```
Comment (oleksandr-danylchenko) [inline frontend/src/common/types/form/form-control-path.type.ts]: Why did you make this custom type?

File: frontend/src/common/types/form/form-control-errors.type.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { FieldErrors } from 'react-hook-form';
+
+type FormControlErrors = FieldErrors;
+
+export { type FormControlErrors };
```
Comment (oleksandr-danylchenko) [inline frontend/src/common/types/form/form-control-errors.type.ts]: Why it's reexported?

File: frontend/src/common/types/form/form-control.type.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { Control } from 'react-hook-form';
+
+type FormControl = Control;
+
+export { type FormControl };
```
Comment (oleksandr-danylchenko) [inline frontend/src/common/types/form/form-control.type.ts]: Why reexport?

File: frontend/src/components/common/input-field/input-field.tsx [hunk 5]
```diff
@@ -9,30 +12,41 @@ import { clsx } from 'clsx';
 
 import styles from './styles.module.scss';
 
-export const InputField = (props: InputFieldPropsType): React.ReactElement => {
+export const InputField = ({
+  name,
+  control,
+  type,
+  errors,
+}: InputFieldPropsType): React.ReactElement => {
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/common/input-field/input-field.tsx]: Please use the `React`'s `FC` type for the components declaration:

```jsx
interface SomeProps { foo: boolean };
const  SomeComponent: FC<SomeProps> = ({ foo }) => {...}
```

File: frontend/src/helpers/validation/get-form-validation-resolver.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { Resolver } from 'react-hook-form';
+
+import { ValidationSchema } from '@common/types/types';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+const getFormValidationResolver = (
+  validationSchema: ValidationSchema<object>,
+): Resolver => {
+  return yupResolver(validationSchema);
+};
```
Comment (oleksandr-danylchenko) [inline frontend/src/helpers/validation/get-form-validation-resolver.ts]:
```suggestion
const getFormValidationResolver = (
  validationSchema: ValidationSchema<object>,
): Resolver => yupResolver(validationSchema);
```

File: frontend/src/store/queries/index.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+} from '@reduxjs/toolkit/query/react';
+
+type ErrorType = {
+  data: {
+    message: string;
+  };
+  status: number;
+};
+
+export const Api = createApi({
```
Comment (oleksandr-danylchenko) [inline frontend/src/store/queries/index.ts]: We should use the names that start in uppercase. The `camelCase` is the standard

File: shared/src/common/types/validation/validation-schema.type.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+export { type SchemaOf as ValidationSchema } from 'yup';
```
Comment (oleksandr-danylchenko) [inline shared/src/common/types/validation/validation-schema.type.ts]: Why reexport?

**Review by oleksandr-danylchenko — CHANGES_REQUESTED** [review]
Some polishing would be appreciated

**Review by nikita-remeslov — DISMISSED** [review]
Looks good to me.
PS. Please resolve conflicts

**Review by oleksandr-danylchenko — DISMISSED** [review]

File: frontend/yarn.lock [hunk 9]
```diff
@@ -1299,10 +1299,17 @@ react-dom@18.2.0:
...
+react-hook-form@^7.34.1:
+  version "7.34.1"
+  resolved "https://registry.yarnpkg.com/react-hook-form/-/react-hook-form-7.34.1.tgz#06cb216daf706bf9ae4969747115afae0d09410d"
+  integrity sha512-tH7TaZgAURMhjzVE2M/EFmxHz2HdaPMAVs9FXTweNW551VlhXSuVcpcYlkiMZf2zHQiTztupVFpBHJFTma+N7w==
+=======
 react-hook-form@^7.34.2:
   version "7.34.2"
   resolved "https://registry.yarnpkg.com/react-hook-form/-/react-hook-form-7.34.2.tgz#9ac6d1a309a7c4aaa369d1269357a70e9e9bf4de"
   integrity sha512-1lYWbEqr0GW7HHUjMScXMidGvV0BE2RJV3ap2BL7G0EJirkqpccTaawbsvBO8GZaB3JjCeFBEbnEWI1P8ZoLRQ==
+>>>>>>> develop
```
Comment (oleksandr-danylchenko) [inline frontend/yarn.lock]:
```suggestion
react-hook-form@^7.34.2:
  version "7.34.2"
  resolved "https://registry.yarnpkg.com/react-hook-form/-/react-hook-form-7.34.2.tgz#9ac6d1a309a7c4aaa369d1269357a70e9e9bf4de"
  integrity sha512-1lYWbEqr0GW7HHUjMScXMidGvV0BE2RJV3ap2BL7G0EJirkqpccTaawbsvBO8GZaB3JjCeFBEbnEWI1P8ZoLRQ==
```

**Review by oleksandr-danylchenko — APPROVED** [review]

**Pull request merged**